sklearn/utils/_encode.py

```diff
@@ -42,9 +133,7 @@ def _unique_python(values, *, return_inverse):
                         f"strings or numbers. Got {types}")
 
     if return_inverse:
-        table = {val: i for i, val in enumerate(uniques)}
-        inverse = np.array([table[v] for v in values])
-        return uniques, inverse
+        return uniques, _map_to_integer(values, uniques)
```

Comment:
`_extract_missing` above will treat None as missing, but I don't think `_map_to_integer` does. Is this problematic? If we are allowing a mixed use of NaN and None then I suspect `_nandict` needs a couple of extra `or key is None`.

Reply:
Since `None` can be used as a key in a normal dictionary. This would treat `None` and `nan` as two different categories. Do you think this behavior is surprising?

Reply:
It does simplify things we error in the situations when there is `nan` and `None`.

Reply:
As of now, I think we can support the weird case of `nan` and `None` together.